File: datasets_v2/communication/ChatSecureAndroid/compose/passphrase_view.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun passphrase_view() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
    Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.TopCenter) {
    Text(text = "Passphrase Setup", modifier = Modifier.fillMaxWidth().wrapContentHeight(), color = Color(0xFFFFFFFF), fontSize = TextUnit.Unspecified)
    }
    Column(modifier = Modifier.fillMaxHeight().wrapContentWidth()) {
      Row(modifier = Modifier.weight(1.0f)) {
        Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().verticalScroll(rememberScrollState())) {
          Column(modifier = Modifier.fillMaxHeight().wrapContentWidth()) {
            Row(modifier = Modifier.weight(1.0f)) {
              Text(text = "Before you get started please choose a secure passphrase to protect your ChatSecure data from unjust access.", modifier = Modifier.wrapContentWidth().wrapContentHeight())
            }
            Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
              Text(text = "Passphrase", modifier = Modifier.wrapContentHeight().width(100.0.dp))
            }
            Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
              OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.size(200.0.dp, 30.0.dp))
            }
            Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
              Text(text = "Passphrase (again)", modifier = Modifier.wrapContentHeight().width(100.0.dp))
            }
            Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
              OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.size(200.0.dp, 30.0.dp))
            }
          }
        }
      }
      Row(modifier = Modifier.fillMaxWidth().height(30.0.dp)) {
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Text(text = "button1")
        }
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Text(text = "button2")
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun passphrase_view_preview() {
    MyApplicationTheme {
        passphrase_view()
    }
}

File: experiments/communication/ChatSecureAndroid/swiftui/Passphrase_view.swift
import SwiftUI

struct Passphrase_view: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      Text("Passphrase Setup").foregroundColor(Color(red: 1, green: 1, blue: 1, opacity: 1)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity, alignment: .center)
      VStack(alignment: .leading, spacing: 0) {
        HStack(alignment: .center, spacing: 0) {
          ScrollView {
            VStack(alignment: .leading, spacing: 0) {
              HStack(alignment: .center, spacing: 0) {
                Text("Before you get started please choose a secure passphrase to protect your ChatSecure data from unjust access.")
              }
              .frame(maxWidth: .infinity, maxHeight: .infinity)
              HStack(alignment: .center, spacing: 0) {
                Text("Passphrase").frame(width: 100.0)
              }
              HStack(alignment: .center, spacing: 0) {
                TextField("", text: .constant("")).frame(width: 200.0, height: 30.0)
              }
              HStack(alignment: .center, spacing: 0) {
                Text("Passphrase (again)").frame(width: 100.0)
              }
              HStack(alignment: .center, spacing: 0) {
                TextField("", text: .constant("")).frame(width: 200.0, height: 30.0)
              }
            }
            .frame(maxHeight: .infinity)
          }
          .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        HStack(alignment: .center, spacing: 0) {
          Button(action: { }) {
            Text("button1")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
          Button(action: { }) {
            Text("button2")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
        }
        .frame(maxWidth: .infinity)
        .frame(height: 30.0)
      }
      .frame(maxHeight: .infinity)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Passphrase_view_Previews: PreviewProvider {
  static var previews: some View {
    Passphrase_view()
  }
}
